airflow/www/views.py

```diff
@@ -678,18 +678,19 @@ def index(self):
                 for name, in dagtags
             ]
 
-            import_errors = session.query(errors.ImportError).order_by(errors.ImportError.id).all()
+            import_errors = session.query(errors.ImportError).order_by(errors.ImportError.id)
 
-        if import_errors:
-            dag_filenames = {dag.fileloc for dag in dags}
-            all_dags_readable = (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG) in user_permissions
+            if (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG) not in user_permissions:
+                # if the user doesn't have access to all DAGs, only display errors from visible DAGs
+                import_errors = import_errors.join(
+                    DagModel, DagModel.fileloc == errors.ImportError.filename
+                ).filter(DagModel.dag_id.in_(filter_dag_ids))
 
-            for import_error in import_errors:
-                if all_dags_readable or import_error.filename in dag_filenames:
-                    flash(
-                        "Broken DAG: [{ie.filename}] {ie.stacktrace}".format(ie=import_error),
-                        "dag_import_error",
-                    )
+        for import_error in import_errors:
+            flash(
+                "Broken DAG: [{ie.filename}] {ie.stacktrace}".format(ie=import_error),
+                "dag_import_error",
+            )
```

Comment:
```suggestion
            for import_error in import_errors:
                flash(
                    "Broken DAG: [{ie.filename}] {ie.stacktrace}".format(ie=import_error),
                    "dag_import_error",
                )
```

I'm thinking we now want to be in the create_session contextmanager now that we are lazily iterating?

Reply:
Does it functionally make a difference? Since the query is instantiated with the session within the context manager it would still be using the same session outside of the context manager, right?

I agree that it makes sense to move it over, I'm just trying to understand why this isn't breaking currently 🤔

Reply:
Basically, `create_session` ends up calling `commit` then `close` (see [create_session source](https://github.com/apache/airflow/blob/eebfeec4fa798b728141dfdeb4cf70970c71067f/airflow/utils/session.py#L27-L37) but `close` should more or less be considered `reset` (see [sqlalchemy docs](https://docs.sqlalchemy.org/en/14/orm/session_basics.html#closing):

> When the Session is closed, it is essentially in the original state as when it was first constructed, and may be used again.

So by using the session again outside the context manager, nothing is doing "proper" cleanup on it at that point. Not a huge deal in this specific case, but 🤷‍♂️.